Reject requests with large page offset

jtgeibel opened the pull request:
`WEB_PAGE_OFFSET_UA_BLOCKLIST` is a comma seperated list of user-agents
that will be blocked if `WEB_MAX_ALLOWED_PAGE_OFFSET` is exceeded.

`WEB_MAX_ALLOWED_PAGE_OFFSET` defaults to 200.
`WEB_PAGE_OFFSET_UA_BLOCKLIST` defaults to the empty list. Because
empty user-agents are already blocked for non-download endpoints, the
check is disabled if this variable is empty or undefined.

pietroalbini commented:
Hmm, my worry with this PR is that we would also reject clients complying with our crawlers policy.

jtgeibel commented:
Sorry for the delay on this. I've added a 2nd environment variable to block specific user-agents only. I've dropped the frontend commit, and have updated the PR description.

pietroalbini commented:
@bors r+

Thanks!

bors commented:
:pushpin: Commit 22ead04410dd8ed5137f3ddaa0b332f332b7dce4 has been approved by `pietroalbini`

<!-- @bors r=pietroalbini 22ead04410dd8ed5137f3ddaa0b332f332b7dce4 -->
<!-- homu: {"type":"Approved","sha":"22ead04410dd8ed5137f3ddaa0b332f332b7dce4","approver":"pietroalbini"} -->

bors commented:
:hourglass: Testing commit 22ead04410dd8ed5137f3ddaa0b332f332b7dce4 with merge 988efc1c8a808e6f24150d192c36686f603730a5...
<!-- homu: {"type":"BuildStarted","head_sha":"22ead04410dd8ed5137f3ddaa0b332f332b7dce4","merge_sha":"988efc1c8a808e6f24150d192c36686f603730a5"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang/crates.io/runs/2629299467)
Approved by: pietroalbini
Pushing 988efc1c8a808e6f24150d192c36686f603730a5 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"pietroalbini","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang/crates.io/runs/2629299467"},"merge_sha":"988efc1c8a808e6f24150d192c36686f603730a5"} -->
